core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -133,10 +156,29 @@ public DefaultTopologyMonitor(InternalDriverContext context) {
     savePort(channel);
 
     CompletionStage<AdminResult> localQuery = query(channel, "SELECT * FROM system.local");
-    CompletionStage<AdminResult> peersQuery = query(channel, "SELECT * FROM system.peers");
+    CompletionStage<AdminResult> peersV2Query = query(channel, "SELECT * FROM system.peers_v2");
+    CompletableFuture<AdminResult> peersQ = new CompletableFuture<>();
+
+    peersV2Query.whenComplete(
+        (r, t) -> {
+          if (t != null) {
+            // The query to system.peers_v2 failed, whe should not attempt this query in the future.
+            // Since this method should be run as part of the control connection initialization, we
+            // shouldn't
+            // need this fallback anywhere else.
+            this.isSchemaV2 = false;
```

Comment:
There is an edge case where it would be done after connection initialization: if we're in a mixed 3.x/4.0 cluster and the first contact point was a 4.0 node, `isSchemaV2` is set when we first connect.
If the control connection later drops and reconnects to a 3.x node, it will reach this line and flip the flag back to false. But the driver recovers correctly: the first thing we do on a control connection reconnect is a `refreshNodeList`, so there's no risk of running a `refreshNode` with the flag still set to true. And if we're in a mixed cluster, all nodes should still be using the same ports, so it doesn't make any difference whether we use `peers` or `peers_v2` after the reconnection.

So I don't think we need to change anything to handle that case.

Reply:
agree, also in a mixed version cluster, the user shouldn't be setting up multiple nodes on the same interface, so I agree that it's ok to drop back to the `peers` table permanently.